runtime/grid_sync.cu

```diff
@@ -150,4 +150,82 @@ __device__ void sync(
   block_sync::sync<Aligned>();
 }
 
+// Non-blocking function to acquire the semaphore value in each calling thread
+__device__ int64_t semaphoreFetch(int64_t* semaphore) {
+  int64_t state;
+#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 700
+  asm volatile("ld.global.acquire.gpu.b64 %0, [%1];\n"
+               : "=l"(state)
+               : "l"(semaphore));
+#else
+  asm volatile("ld.global.cg.b64 %0, [%1];\n" : "=l"(state) : "l"(semaphore));
```

Comment:
Does this need to be an inline PTX code?

Reply:
Not necessarily. It also is only active for pre-Volta, so we IIUC we are not really supporting this code path. The reason for inline PTX here is the `.cg` qualifier, equivalent to the runtime functions that use `CacheOp::Global`. We can get a similar effect by passing the semaphore as `volatile` though I think.